_docs/tasks/security/authn-policy.md

```diff
@@ -0,0 +1,300 @@
+
+* Know how to verify mTLS setup (recommend to walk through [testing Istio mutual TLS authentication]({{home}}/docs/tasks/security/mutual-tls.html))
+
+* Have a Kubernetes cluster with Istio installed, without mTLS. See [the Istio installation task]({{home}}/docs/setup/kubernetes/quick-start.html) and follow step 5a.
+
+* For demo, create two namespaces `foo` and `bar`, and deploy [httpbin](https://github.com/istio/istio/tree/master/samples/httpbin) and [sleep](https://github.com/istio/istio/tree/master/samples/sleep) with sidecar on both of them. Also, run another sleep app without sidecar (to keep it separate, run it in `legacy` namespace)
+
+```bash
+kubectl create ns foo
+kubectl apply -f <(istioctl kube-inject -f samples/httpbin/httpbin.yaml) -n foo
+kubectl apply -f <(istioctl kube-inject -f samples/sleep/sleep.yaml) -n foo
+
+kubectl create ns bar
+kubectl apply -f <(istioctl kube-inject -f samples/httpbin/httpbin.yaml) -n bar
+kubectl apply -f <(istioctl kube-inject -f samples/sleep/sleep.yaml) -n bar
+
+kubectl create ns legacy
+kubectl apply -f samples/sleep/sleep.yaml -n legacy
+
+```
+
+* Verifying setup by sending an http request (using curl command) from any sleep pod (among those in namespace `foo`, `bar` or `legacy`) to either `httpbin.foo` or `httpbin.bar`. All requests should success with http code 200.
+
+For example, here is a command to check `sleep.bar` to `httpbin.foo` reachability:
+
+```bash
+kubectl exec $(kubectl get pod -l app=sleep -n bar -o jsonpath={.items..metadata.name}) -c sleep -n bar -- curl http://httpbin.foo:8000/ip -s -o /dev/null -w "%{http_code}\n"
+```
+```
+200
+```
+
+Conveniently, this one-liner command iterates through all combinations:
+
+```bash
+for from in "foo" "bar" "legacy"; do for to in "foo" "bar"; do kubectl exec $(kubectl get pod -l app=sleep -n ${from} -o jsonpath={.items..metadata.name}) -c sleep -n ${from} -- curl http://httpbin.${to}:8000/ip -s -o /dev/null -w "sleep.${from} to httpbin.${to}: %{http_code}\n"; done; done
+```
+
+```
+sleep.foo to httpbin.foo: 200
+sleep.foo to httpbin.bar: 200
+sleep.bar to httpbin.foo: 200
+sleep.bar to httpbin.bar: 200
+sleep.legacy to httpbin.foo: 200
+sleep.legacy to httpbin.bar: 200
+```
+
+
+* Also verify that there are no authencation policy in the system
+
+```bash
+kubectl get policies.authentication.istio.io -n foo
+
+kubectl get policies.authentication.istio.io -n bar
+```
+
+```
+No resources found.
+```
+
+## Enable mTLS for all services in namespace `foo`
+
+Run this command to set namespace-level policy for namespace `foo`.
+
+```bash
+cat <<EOF | istioctl create -f -
+apiVersion: "authentication.istio.io/v1alpha1"
+kind: "Policy"
+metadata:
+  name: "enable-mtls"
+  namespace: "foo"
+spec:
+  peers:
+  - mtls:
+EOF  
+```
+
+And verify policy is added:
```

Comment:
verify policy added -> verify the policy was added:

Reply:
Done